src/Mobs/AggressiveMonster.h

```diff
@@ -30,6 +30,13 @@ class cAggressiveMonster:
 
 	virtual void EventSeePlayer(cPlayer * a_Player, cChunk & a_Chunk) override;
 
+	/**
+	* Check if a a_mobtype is around the monster at a SightDistance
+	* return true if it found one and set a_Monster as the Found Monster
+	* return false and leave the a_Monster untouched if
+	*/
+	virtual bool CanSeeMobType(eMonsterType a_mobtype, cAggressiveMonster * a_Monster, int SightDistance=16);
+
```

Comment:
```suggestion
	/** Returns a mob of the specified type that is visible to this monster within the specified sight distance.
	If there are multiple such monsters, returns a non-specified (random) one of them. 
	Returns nullptr if there is no such monster. */
	virtual cMonster * GetMonsterOfTypeInSight(eMonsterType a_MobType, double SightDistance=16);

```

Reply:
Also, this might be better implemented directly in the `cMonster` class and probably doesn't need to be `virtual` at all.